### PR TITLE
RES_NET_ARP_ALL changes and QA infra.

### PR DESCRIPTION
**Pull request by anjalidk**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 CC = gcc
-CFLAGS = -g -Wall -Werror -fPIC -std=gnu99
+CFLAGS = -g -Wall -Werror -fPIC -std=gnu99 -DTESTING
```
Comment (hallyn) [inline Makefile:2]: Do you actually want -DTESTING unconditionally enabled in 'make' ?  or was this meant to stay in a testing branch?

File: tests/ARP/arp.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+export LD_LIBRARY_PATH=`git rev-parse --show-toplevel`
+cd $LD_LIBRARY_PATH
+cd tests/ARP
+rm -f ./arp_test
+cc -I $LD_LIBRARY_PATH -std=gnu99 -o arp_test arp_test.c -L $LD_LIBRARY_PATH -lresource
+rm -f ./arp_info.orig
+rm -f ./arp_info.txt
+arp -n -a > arp_info1.orig
+#remove all "? " from above file
+cat arp_info1.orig | sed 's/? //g' > arp_info.orig
```
Comment (hallyn) [inline tests/ARP/arp.sh:11]: Hm - you really can't count on 'sed -i' being available?

**Review by hallyn — CHANGES_REQUESTED** [review]

**Comment by hallyn** [comment]
> I can add a note to README saying sed needs to be installed if testing mode needs to be enabled?

No, I'm saying you appear to be doing
```
cat file1 | sed 's/.../' > file2; mv file2 file1
```
instead of just
```
sed -i 's/.../' file1
```

**Comment by anjalidk** [comment]
> On Aug 14, 2023, at 11:31 AM, Serge Hallyn ***@***.***> wrote:
> 
> 
> I can add a note to README saying sed needs to be installed if testing mode needs to be enabled?
> 
> No, I'm saying you appear to be doing
> 
> cat file1 | sed 's/.../' > file2; mv file2 file1
> 
> instead of just
> 
> sed -i 's/.../' file1

Yes, thanks done. Pushed with review comments amended to previous commit.

Anjali

File: net_arp.c [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+			printf("ndm_flags: %u\n",nd_msg->ndm_flags);
+			printf("ndm_state: %u\n", nd_msg->ndm_state);
+#endif
+			len -= NLMSG_LENGTH(sizeof(*nd_msg));
+			if (len < 0) {
+				eprintf("nlmsg_len incorrect");
+				errno = EINVAL;
+				free(buf);
+				free(arps);
+				return -1;
```
Comment (hallyn) [inline net_arp.c:269]: You left the 'return -1' above the 'ret = -1; goto out;'

**Comment by anjalidk** [comment]
> On Aug 15, 2023, at 7:07 PM, Serge Hallyn ***@***.***> wrote:
> 
> 
> @hallyn commented on this pull request.
> 
> In net_arp.c:
> 
> > +#ifdef PRINTLOGS
> +			printf("r->nlmsg_type is %d (RTM_NEWNEIGH = 28)\n", r->nlmsg_type);
> +			printf("Received ndmsg, len %d\n", len);
> +			printf("ifindex is %d\n", nd_msg->ndm_ifindex);
> +			printf("family: %u\n", nd_msg->ndm_family);
> +			printf("ndm_type: %u\n", nd_msg->ndm_type);
> +			printf("ndm_flags: %u\n",nd_msg->ndm_flags);
> +			printf("ndm_state: %u\n", nd_msg->ndm_state);
> +#endif
> +			len -= NLMSG_LENGTH(sizeof(*nd_msg));
> +			if (len < 0) {
> +				eprintf("nlmsg_len incorrect");
> +				errno = EINVAL;
> +				free(buf);
> +				free(arps);
> +				return -1;
> 
> You left the 'return -1' above the 'ret = -1; goto out;’
> 
Sorry! Committed with return -1 removed.

Anjali

File: net_arp.c [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+	struct msghdr msg;
+	struct nlmsghdr *r;
+	struct nlmsgerr *nlmsg_err;
+	struct ndmsg *nd_msg;
+	struct rtattr *at[NDA_MAX + 1], *rta;
+	struct arp_info *arps = NULL, *iarps = NULL;
+	struct sockaddr_nl nladdr;
+#ifdef TESTING
+	FILE *fp = NULL;
+	fp = fopen ("./arp_info.txt", "w");
```
Comment (hallyn) [inline net_arp.c:178]: Now, this is just testing code, so probably no big deal, but you're not checking fopen return value

Comment (hallyn) [reply net_arp.c:178]: If you're around and push a fix in the next 5m, great - else i'll just merge it as is for now.

**Comment by anjalidk** [comment]
> On Aug 15, 2023, at 7:29 PM, Serge Hallyn ***@***.***> wrote:
> 
> 
> @hallyn commented on this pull request.
> 
> In net_arp.c:
> 
> > +
> +static int handle_arp_resp(int net_sock, void **out)
> +{
> +	int err, recvl, len, max_arp, arp_size, ret = 0;
> +	int aind = 0, total_arp_size = 0;
> +	struct iovec iov;
> +	struct msghdr msg;
> +	struct nlmsghdr *r;
> +	struct nlmsgerr *nlmsg_err;
> +	struct ndmsg *nd_msg;
> +	struct rtattr *at[NDA_MAX + 1], *rta;
> +	struct arp_info *arps = NULL, *iarps = NULL;
> +	struct sockaddr_nl nladdr;
> +#ifdef TESTING
> +	FILE *fp = NULL;
> +	fp = fopen ("./arp_info.txt", "w");
> 
> If you're around and push a fix in the next 5m, great - else i'll just merge it as is for now.
> 
Yup, I will push shortly..

Anjali

**Comment by anjalidk** [comment]
> On Aug 15, 2023, at 7:37 PM, Anjali Kulkarni ***@***.***> wrote:
> 
> 
> 
>> On Aug 15, 2023, at 7:29 PM, Serge Hallyn ***@***.***> wrote:
>> 
>> 
>> @hallyn commented on this pull request.
>> 
>> In net_arp.c:
>> 
>>> +
>> +static int handle_arp_resp(int net_sock, void **out)
>> +{
>> +	int err, recvl, len, max_arp, arp_size, ret = 0;
>> +	int aind = 0, total_arp_size = 0;
>> +	struct iovec iov;
>> +	struct msghdr msg;
>> +	struct nlmsghdr *r;
>> +	struct nlmsgerr *nlmsg_err;
>> +	struct ndmsg *nd_msg;
>> +	struct rtattr *at[NDA_MAX + 1], *rta;
>> +	struct arp_info *arps = NULL, *iarps = NULL;
>> +	struct sockaddr_nl nladdr;
>> +#ifdef TESTING
>> +	FILE *fp = NULL;
>> +	fp = fopen ("./arp_info.txt", "w");
>> 
>> If you're around and push a fix in the next 5m, great - else i'll just merge it as is for now.
>> 
> Yup, I will push shortly..

Push is done.
Anjali

**Review by hallyn — APPROVED** [review]

**Pull request merged**